### DISCUSSION TITLE
Raw query params when using ".execute": is there a way to properly escape some chars

### DISCUSSION THREAD

**Discussion by haykerman** [mikro-orm/mikro-orm#4099] [Q&A] [answered]
#### This question is regarding the `this.execute` function in a migration.

Take a look at this example:
`await this.execute('insert into "logs" ("name") values (\'Hello, it\'s Hayk\')');`

Is there a way to automatically escape single quotes when running a raw insert query with `.execute`? Or more generally, is there way to pass parameters into raw query instead of concatenating the string? And then that param will automatically get escaped/transformed properly.

Example: using [named params with pg-promise](https://github.com/vitaly-t/pg-promise#sql-names)

**Comment by B4nan** [comment, marked as answer]
That method has a second `params` parameter:

```ts
await em.execute('insert into "logs" ("name") values (?)', ["Hello, it's Hayk"]);
```

> This question is regarding the this.execute function in a migration.

Then I dont understand why your example is using `em` :] This is an omission, I'll add the params there too. But you can already do this:

```ts
await this.driver.execute('...query...', [...params...], 'all', this.ctx);
```

**Reply by haykerman** [reply]
Sorry for the typo in the example. Fixed it.

Thanks for the answer. Glad to hear the support for params will be added. It'd be much appreciated.
Will be using the `driver.execute` one for now.

BTW, what's the main difference between `this.execute` and `this.driver.execute`. Is it the "context awareness"? Like we don't need to pass context to `this.execute` one.

**Reply by B4nan** [reply]
https://github.com/mikro-orm/mikro-orm/blob/master/packages/migrations/src/Migration.ts#L37-L39
